nomad/node_endpoint.go

```diff
@@ -890,7 +898,7 @@ func (n *Node) GetNode(args *structs.NodeSpecificRequest,
 	if err != nil {
 		return err
 	}
-	if !aclObj.AllowNodeRead() {
+	if aclObj != nil && !aclObj.AllowNodeRead() {
```

Comment:
Huh, how did this sneak by? I would think it would panic on `nomad node status ...` for everyone with ACLs disabled?

Reply:
Yeah that burned me here... the old `ResolveToken` code returns `nil, nil` in that case, so we just skip over the `AllowNodeRead` check. Fortunately we have good unit test coverage of non-ACLs vs with-ACLs and that caught it.